util/xxhash3/hash.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2021 ByteDance Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package xxhash3 implements https://github.com/Cyan4973/xxHash/blob/dev/xxhash.h
+package xxhash3
+
+import (
+	"math/bits"
+	"unsafe"
+
+	"github.com/bytedance/gopkg/internal/runtimex"
+)
+
+// Hash returns the hash value of the byte slice in 64bits.
+func Hash(data []byte) uint64 {
+	funcPointer := hashSmall
+
+	if len(data) > 16 {
+		funcPointer = hashLarge
+	}
+	return hashfunc[funcPointer](*(*unsafe.Pointer)(unsafe.Pointer(&data)), len(data))
+
+}
+
+// HashString returns the hash value of the string in 64bits.
+func HashString(s string) uint64 {
+	return Hash([]byte(s))
```

Comment:
It is better to use `https://github.com/bytedance/gopkg/blob/develop/internal/hack/hack.go#L27` instead of `[]byte`